src/core/client/stream/tabs/Comments/Comment/CommentContainer.tsx

```diff
@@ -82,438 +89,421 @@ interface Props {
   toggleCollapsed?: () => void;
 }
 
-interface State {
-  showReplyDialog: boolean;
-  showEditDialog: boolean;
-  editable: boolean;
-  showReportFlow: boolean;
-}
-
-export class CommentContainer extends Component<Props, State> {
-  private uneditableTimer: any;
-
-  public state = {
-    showReplyDialog: false,
-    showEditDialog: false,
-    editable: this.isEditable(),
-    showReportFlow: false,
-  };
+export const CommentContainer: FunctionComponent<Props> = ({
+  className,
+  collapsed,
+  comment,
+  disableReplies,
+  hideAnsweredTag,
+  hideModerationCarat,
+  highlight,
+  indentLevel,
+  localReply,
+  onRemoveAnswered,
+  settings,
+  showConversationLink,
+  hideReportButton,
+  story,
+  toggleCollapsed,
+  eventEmitter,
+  setCommentID,
+  viewer,
+  showAuthPopup,
+}) => {
+  const [showReplyDialog, setShowReplyDialog] = useState(false);
+  const [
+    showEditDialog,
+    setShowEditDialog,
+    toggleShowEditDialog,
+  ] = useToggleState(false);
+  const [showReportFlow, , toggleShowReportFlow] = useToggleState(false);
+  const handleShowConversation = useCallback(
+    (e: MouseEvent) => {
+      ViewConversationEvent.emit(eventEmitter, {
+        commentID: comment.id,
+        from: "COMMENT_STREAM",
+      });
+      e.preventDefault();
 
-  constructor(props: Props) {
-    super(props);
-    if (this.isEditable()) {
-      this.uneditableTimer = this.updateWhenNotEditable();
-    }
-  }
+      void setCommentID({ id: comment.id });
 
-  public componentWillUnmount() {
-    clearTimeout(this.uneditableTimer);
-  }
+      return false;
+    },
+    [eventEmitter, comment, setCommentID]
```

Comment:
instead of comment, here it's more optimised to set `comment.id` into the dependency array.